### PR TITLE
improve: register self with `__file__`

### PR DESCRIPTION
**Pull request by Borda**
replace parsing path from import and rather use `__file__` which is widely supported and more robust reference with respect to various installing styles as for example `pip install -e .`

cc: @garrettj403 @echedey-ls

### CODE REVIEW

File: scienceplots/__init__.py [hunk 1]
```diff
@@ -1,11 +1,8 @@
 import os  # pathlib.Path.walk not available in Python <3.12
 import matplotlib.pyplot as plt
 
-import scienceplots
-
 # register the bundled stylesheets in the matplotlib style library
-scienceplots_path = scienceplots.__path__[0]
-styles_path = os.path.join(scienceplots_path, 'styles')
+styles_path = os.path.join(__file__, 'styles')
```
Comment (echedey-ls) [inline scienceplots/__init__.py:5]: `__file__` is resolving to `scienceplots/__init__.py`, so `styles_path = os.path.join(__file__, 'styles')`  resolves to `scienceplots/__init__.py/styles`, which is undesirable.

**Review by echedey-ls — COMMENTED** [review]
Can you elaborate on what this PR tries to do? Have you encountered any problems with editable installs? They work for me on Win10, and I'm 99% confident I've also tested them on linux long ago.

File: scienceplots/__init__.py (hunk 1 above)
Comment (Borda) [reply scienceplots/__init__.py:5]:
```suggestion
styles_path = os.path.join(os.path.dirname(__file__), 'styles')
```

**Comment by echedey-ls** [comment]
I still need to know / have an example / see an official statement of preference on `__file__` vs `__path__[0]` to understand what this PR fixes. IMO, code at master has two pros compared to this PR:
1. In case code is being executed inside an `eval()`, the current code resolves correctly to the folder with the styles. This is an edge-case that I would not expect to happen ever, but the possibility is there in case we need to test by doing that.
2. I find master's code slightly more readable than this PR, without a chain of `os.path. ...` calls one inside the other. But that's super-negligible.

I hope you understand that rewriting code unless it fixes issues / addresses future problems is not actually desirable. You can close the PR if no further work is expected to be done.

There are still tasks and other aspects in which SciencePlots can be improved, like tests and documentation (I have plans and have already started working on it). Let me know if you would like to help with that, so maybe we can both work on it.

**Comment by echedey-ls** [comment]
Again, thanks for the interest you showed in this project @Borda ❣️  . If you wish to help with other tasks, feel free to contact me.

I will close this PR for the mentioned reasons.

**Comment by Borda** [comment]
> If you wish to help with other tasks, feel free to contact me.

Sure, if you have some low hanging fruits =) 

> I will close this PR for the mentioned reasons.

Right, I had this problem with some other projects, but not yet with this one in particular...

**Comment by echedey-ls** [comment]
I will let you know once I have something more evolved :D 
I must admit that it will take me some time thou.
